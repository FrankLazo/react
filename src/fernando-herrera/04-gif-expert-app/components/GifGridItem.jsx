import PropTypes from 'prop-types';
import 'animate.css';

const GifGridItem = ({ title, url }) =>
{
	return (
		<div className="card mb-4 mr-4 animate__animated animate__fadeIn">
			<div className="card-image">
				<figure className="image m-0">
					<img src={ url } alt={ title } />
				</figure>
			</div>

			<div className="card-content">
				<div className="content">
					<p >{ title }</p>
				</div>
			</div>
		</div>
	);
};

GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	url:   PropTypes.string.isRequired,
}

export default GifGridItem;
