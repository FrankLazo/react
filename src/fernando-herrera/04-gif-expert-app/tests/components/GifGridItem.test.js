import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Gif Grid Item', () => {
	const title = 'Image title';
	const url   = 'https://localhost/something.jpg';
	const wrapper = shallow(<GifGridItem title={title} url={url} />);

	test('should show right component <GifGridItem />', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should have a paragraph with title', () => {
		const paragraph = wrapper.find('p');
		expect(paragraph.text().trim()).toBe(title);
	});

	test('should have a image with url and title', () => {
		const image = wrapper.find('img');
		expect(image.prop('src')).toBe(url);
		expect(image.prop('alt')).toBe(title);
	});

	test('should have animate__fadeIn', () => {
		const div = wrapper.find('.card');

		// const className = div.prop('className');
		// expect(className.includes('animate__fadeIn')).toBe(true);

		expect(div.hasClass('animate__fadeIn')).toBe(true);
	});
});
