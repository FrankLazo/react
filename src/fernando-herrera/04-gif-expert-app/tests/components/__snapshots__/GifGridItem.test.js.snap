// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Grid Item should show right component <GifGridItem /> 1`] = `
<div
  className="card mb-4 mr-4 animate__animated animate__fadeIn"
>
  <div
    className="card-image"
  >
    <figure
      className="image m-0"
    >
      <img
        alt="Image title"
        src="https://localhost/something.jpg"
      />
    </figure>
  </div>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <p>
        Image title
      </p>
    </div>
  </div>
</div>
`;
